spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java

```diff
@@ -0,0 +1,799 @@
+        ProductDTO prod5 = ownerProductApi.createProductByOwner(owner2.getKey(), Products.random());
+        ProductDTO prod5d = ownerProductApi.createProductByOwner(owner2.getKey(), Products.random());
+        ProductDTO prod6 = ownerProductApi.createProductByOwner(owner3.getKey(), Products.random());
+        ProductDTO prod6d = ownerProductApi.createProductByOwner(owner3.getKey(), Products.random());
+
+        PoolDTO pool1 = Pools.random(prod4);
+        pool1.setDerivedProductId(prod4d.getId());
+        pool1.setProvidedProducts(Set.of(new ProvidedProductDTO().productId(product1.getId())));
+        pool1.setDerivedProvidedProducts(Set.of(new ProvidedProductDTO().productId(product2.getId())));
+        ownerApi.createPool(owner1.getKey(), pool1);
+
+        PoolDTO pool2 = Pools.random(prod5);
+        pool2.setDerivedProductId(prod5d.getId());
+        pool2.setProvidedProducts(Set.of(new ProvidedProductDTO().productId(product1.getId()),
+            new ProvidedProductDTO().productId(product2.getId())));
+        pool2.setDerivedProvidedProducts(Set.of(new ProvidedProductDTO().productId(product3.getId())));
+        ownerApi.createPool(owner2.getKey(), pool2);
+
+        PoolDTO pool3 = Pools.random(prod6);
+        pool3.setDerivedProductId(prod6d.getId());
+        pool3.setProvidedProducts(Set.of(new ProvidedProductDTO().productId(product1.getId())));
+        pool3.setDerivedProvidedProducts(Set.of(new ProvidedProductDTO().productId(product3.getId())));
+        ownerApi.createPool(owner3.getKey(), pool3);
+
+        verifyRefreshPoolJob(ownerProductApi, owner1.getKey(), prod4.getId(), true);
+        verifyRefreshPoolJob(ownerProductApi, owner2.getKey(), prod5d.getId(), true);
+        verifyRefreshPoolJob(ownerProductApi, owner3.getKey(), product1.getId(), true);
+        verifyRefreshPoolJob(ownerProductApi, owner3.getKey(), product3.getId(), true);
+
+        assertNotFound(() -> ownerProductApi.refreshPoolsForProduct(owner1.getKey(), "bad_id", true));
+    }
+
+    @Test
+    @DisplayName("should lists all products in bulk fetch")
+    public void shouldListsAllProductsInBulkFetch() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        String ownerKey = owner.getKey();
+        ProductDTO prod1 = ownerProductApi.createProductByOwner(ownerKey, Products.random());
+        ProductDTO prod2 = ownerProductApi.createProductByOwner(ownerKey, Products.random());
+        ProductDTO prod3 = ownerProductApi.createProductByOwner(ownerKey, Products.random());
+
+        List<ProductDTO> products = ownerProductApi.getProductsByOwner(ownerKey, new ArrayList<>());
+        assertEquals(3, products.size());
+        assertTrue(products.contains(prod1));
+        assertTrue(products.contains(prod2));
+        assertTrue(products.contains(prod3));
+
+        // Pick two products to use in a bulk get
+        products = ownerProductApi.getProductsByOwner(ownerKey, List.of(prod1.getId(), prod3.getId()));
```

Comment:
These should be separate test cases -- one for specifying products and one that doesn't.

Reply:
Updated to two separate tests: shouldListsAllProductsInBulkFetch and shouldListsAllSpecifiedProductsInBulkFetch

Reply:
:+1: